src/Crypto/Macaroon/Verifier/Internal.hs

```diff
@@ -55,6 +55,8 @@ instance Monoid ValidationError where
     _ `mappend` SigMismatch = SigMismatch
     (ValidatorError e) `mappend` (ParseError _) = ValidatorError e
     (ParseError _) `mappend` (ValidatorError e) = ValidatorError e
+    (ValidatorError e1) `mappend` (ValidatorError e2) = ValidatorError $ e1 ++ " " ++ e2
```

Comment:
If you have access to `<>`, replace `++` by `<>`, the same way you'd replace `map` by `fmap` and `return` by `pure`. I think LTS 10.10 exposes `<>` in the Prelude so it should be feasible. If you need to import `<>`, it's not worth it though.